pkg/auth/csp/tanzu.go

```diff
@@ -240,6 +242,21 @@ func (h *cspLoginHandler) runLocalListener(listener net.Listener) func() {
 }
 
 func (h *cspLoginHandler) callbackHandler(w http.ResponseWriter, r *http.Request) {
+	// Lock the mutex to ensure only one request can access/update shared state at a time
+	// Note: This should be remote corner case, but we have seen cases where Chrome browser redirects
+	// and make 2 back to back requests with the same URL before the local server is exited.
+	// In such scenario, the second request would be blocked till the prior request is finished
+	// and then gets unblocked to check if the prior request already acquired token, if so return
+	// the success message else let the request go through for token exchange and return
+	// the response which would fail anyway(context would be canceled for second request)
```

Comment:
This is the best effort to provide the success reponse if the prior request is success(for most scenarios).

Reply:
Does this approach mean the user will see two success messages?
If so, maybe the second time (`if h.token != nil`) we should not print anything?

Reply:
Since I cannot reproduce this issue, I am hoping it will only print only one success message ( I am not sure though) because by the time the second message is done, the first request handler would have called `tokenExchangeComplete()` which would close the local server too (but there is possiblity that second response is written after this method call and server close). 

(While debugging on the system where we saw this issue(not reproducible now anymore), we tried second request checking if the first request is in progress and returning without printing anything(but not with mutex in place) , we saw the browser not showing any text message (the first reponse may not have reached the browser as the second request returned immediately with out printing anything and closed the server).  )

However with mutex inplace I am thinking not printing anything should also make sense (but unfortunately I can't reproduce it and test this theory).  I will update the PR with not printing anything for now, we can revisit this again if we can reproduce it. what do you think?

Reply:
updated the PR with the second request just returning without printing anything.